linkers/gold

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
```

Comment:
seems this file and `linkers/dld` only differ in two lines/variables. Could you make it so that we have one common linker wrapper script, that is parametrized with needed changes?

Reply:
To do this correctly we need to get language as a parameter, (is it GO or JAVA).
We can do this via https://doc.rust-lang.org/cargo/reference/build-scripts.html#rustc-env, in build.rs. I think we should do this even if we will replace this whole area this quarter.